### DISCUSSION TITLE
Where are the bundled styles?

### DISCUSSION THREAD

**Discussion by cjayyy** [easepick/easepick#18] [Q&A]
I am trying to figure out how the styles are working here, but it seems a bit chaotic. In bundled Index there is `./index.scss` import although there is no (s)css file to import. The build looks it doesn't produce any styles at all. Am I searching wrong or is it really missing? 
When the whole tool is such nicely bundled it would be shame to use styles from CDN...

**Comment by wakirin** [comment]
Each package has its own scss file, for example the core package package: https://github.com/easepick/easepick/blob/master/packages/core/src/scss/index.scss

The Bundle package imports from other packages: https://github.com/easepick/easepick/blob/master/packages/bundle/src/index.scss

**Reply by cjayyy** [reply]
I saw these, but those are source files. My problem is that I can't see it in the build files.

Could you please provide an example on how to use built-in (or custom styles from module) styles via modules (and not CDN)?

**Reply by wakirin** [reply]
> I saw these, but those are source files. My problem is that I can't see it in the build files.

Yes, these files don't included in the dist folder.

> Could you please provide an example on how to use built-in (or custom styles from module) styles via modules (and not CDN)?

What Bundler are you using ?
In fact, you can copy the `index.css` from the package into your project and then you don't need CDN link.

**Reply by cjayyy** [reply]
> Yes, these files don't included in the dist folder.

Why not? IMHO it would make sense to include at least the core styles. 

> What Bundler are you using ?

Webpack

> In fact, you can copy the `index.css` from the package into your project and then you don't need CDN link.

That's what I wanted to do as workaround but I didn't find a way how to pass my (source, not built) css file into easepick's `css` option.

**Reply by cjayyy** [reply]
I have found [this way](https://stackoverflow.com/questions/57667688/is-that-possible-to-import-css-file-as-string-in-es6) but that depends on another (loader) package.

**Reply by wakirin** [reply]
> That's what I wanted to do as workaround but I didn't find a way how to pass my (source, not built) css file into easepick's css option.

You don't need to pass SCSS files into `css` options. This option accept only css. You need to build css files first and the pass to the picker.
By the way, check this discussion: https://github.com/easepick/easepick/discussions/8

**Reply by cjayyy** [reply]
OMG, I'm sorry I've missed that discussion 🤦.

In that case I suppose that it is not possible to add custom styles from source file without another loader plugin, right?

**Comment by wakirin** [comment]
> In that case I suppose that it is not possible to add custom styles from source file without another loader plugin, right?

I think I found a solution.
You will need [MiniCssExtractPlugin](https://webpack.js.org/plugins/mini-css-extract-plugin/)

_webpack.config.js_
```js
const path = require('path');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const isProduction = process.env.NODE_ENV == 'production';
// ...
const scss_config = {
  mode: isProduction ? 'production' : 'development',
  entry: [
    './node_modules/@easepick/bundle/dist/index.css', // here import bundle or core styles
    './src/index.scss', // your custom files
  ],
  output: {
    path: path.resolve(__dirname, 'dist'), // path for extract css file
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: 'easepick.css',
      ignoreOrder: false,
    }),
  ],
  module: {
    rules: [
      {
        test: /\.css$/i,
        use: [
          MiniCssExtractPlugin.loader,
          'css-loader',
        ],
      },
      {
        test: /\.s[ac]ss$/i,
        use: [
          MiniCssExtractPlugin.loader,
          'css-loader',
          'sass-loader',
        ],
      },
    ],
  },
};

module.exports = [
  scss_config,
  // your config here,
]
```

_index.ts_
```ts
import { easepick } from '@easepick/bundle';

const picker = new easepick.create({
  element: document.getElementById('datepicker'),
  css: ['./main.css'], // path to created css file
});
```

If you find another solution feel free to share here.

**Reply by cjayyy** [reply]
I'll stick with the `raw-loader` for MVP. 
Later will try this solution. 

Thanks a lot!

**Reply by zipper** [reply]
@wakirin Hi, just found this solution and indeed it works. The only problem I have is, I don't actually know the path - I have a webpack configured to add hash into filename. Is there any way how to keep the filename with hash and in the same time load it here? The correct path is inside generated manifest.json, but I'm not sure if it is even possible to load manifest inside ts file.

Btw the manifest path itself depends not whether the hot module replacement is on (which it is on dev server).
